kernel/tests/read.rs

```diff
@@ -511,8 +513,7 @@ fn read_with_scan_data(
     if expected.is_empty() {
         assert_eq!(batches.len(), 0);
     } else {
-        let schema = batches[0].schema();
```

Comment:
small bug fix here. arrow will assume a column isn't nullable if it doesn't find any nulls in it. so using the schema of the first batch as the schema for all is incorrect if the first batch happens to have a column with no nulls, but subsequent batches DO have nulls. Instead we do the correct thing now and transform the schema the scan promises to read into an arrow schema, and use _that_ as the schema for the batches.